models/base_model_adapter.py

```diff
@@ -41,88 +39,97 @@ def extract_content(line):
             pass
         return ""
 
-    @staticmethod
-    def post_stream(url, data, headers, stream):
-        s = requests.Session()
-        ans = ""
-        response = s.post(url, data=json.dumps(data), headers=headers, stream=stream)
-        if stream:
-            with response:  # To properly closed The response object after the block of code is executed
-                if not response.ok:
-                    raise ValueError(f'error:{response.status_code}, message: {response.text}')
-                logger.info("Streaming the response")
-                for line in response.iter_lines():
-                    if line:
-                        print(line)
-                        line = line.decode('utf-8')
-                        ans += ModelAdapter.extract_content(line)
-        else:
-            ans = response.json().get('choices')[0].get('message').get('content')
-        return ans
+    def post_stream(self, messages):
 
-    def call_model_requests(self, messages):
-        # Configure payload data sending to API endpoint
-        stream = self.model_entity.configuration.get('stream', True)
         data = {
             "messages": messages,
             "max_tokens": self.configuration.get('max_tokens', 1024),
             "temperature": self.configuration.get('temperature', 0.5),
             "top_p": self.model_entity.configuration.get('top_p', 0.7),
-            "stream": stream
+            "stream": self.stream
         }
         headers = {
             "Content-Type": "application/json",
             "Authorization": f"Bearer {self.api_key}"
         }
-        full_answer = self.post_stream(url=self.url, data=data, headers=headers, stream=stream)
 
-        return full_answer
+        s = requests.Session()
+        response = s.post(self.url, data=json.dumps(data), headers=headers, stream=self.stream)
+        if self.stream:
+            try:
+                with response:  # To properly closed The response object after the block of code is executed
+                    if not response.ok:
+                        raise ValueError(f'error:{response.status_code}, message: {response.text}')
+                    logger.info("Streaming the response")
+                    for line in response.iter_lines():
+                        if line:
+                            line = line.decode('utf-8')
+                            yield self.extract_content(line) or ""
+            except requests.exceptions.RequestException as e:
+                logger.warning(f"Request failed: {e}")
```

Comment:
if request failed shouldn't it raise an error?

Reply:
yes, i'll change it